internal/internal_utils.go

```diff
@@ -286,12 +290,12 @@ func getMetricsScopeForLocalActivity(ts *metrics.TaggedScope, workflowType, loca
 	return ts.GetTaggedScope(tagWorkflowType, workflowType, tagLocalActivityType, localActivityType)
 }
 
-func getTimeoutTypeFromErrReason(reason string) s.TimeoutType {
+func getTimeoutTypeFromErrReason(reason string) (s.TimeoutType, error) {
 	timeoutTypeStr := reason[strings.Index(reason, " ")+1:]
 	var timeoutType s.TimeoutType
 	if err := timeoutType.UnmarshalText([]byte(timeoutTypeStr)); err != nil {
-		// this should never happen
-		panic(err)
+		// this happens when the timeout error is constructed by an old client
+		return 0, err
```

Comment:
Return 0 is a little bit weird, usually function should return pointer and in error case return nil pointer.

Reply:
I've seen a lot of code returning 0 and empty strings. IMHO if error is not nil the value doesn't matter at all.